extensions/amp-subscriptions/0.1/local-subscription-platform.js

```diff
@@ -53,6 +56,70 @@ export class LocalSubscriptionPlatform {
         ),
         'Authorization Url'
     );
+
+    // TODO(dvoytenko, #3742): This will refer to the ampdoc once AccessService
+    // is migrated to ampdoc as well.
+    /** @private @const {!Promise<!../../../src/service/cid-impl.Cid>} */
+    this.cid_ = Services.cidForDoc(ampdoc);
+
+    /** @private {UrlBuilder} */
+    this.urlBuilder_ = null;
+
+    /** @private {Actions} */
+    this.actions_ = null;
+
+    /** @private {Promise<string>} */
+    this.readerIdPromise_ = null;
+
+    /** @private {SubscriptionAnalytics} */
+    this.subscriptionAnalytics_ = new SubscriptionAnalytics();
+
+    this.initActions_();
+  }
+
+  /**
+   * Initializes the actions for the local platform
+   * @private
+   */
+  initActions_() {
+    this.urlBuilder_ = new UrlBuilder(this.ampdoc_, this.getReaderId_());
+    user().assert(this.serviceConfig_['actions'],
+        'Actions have not been defined in the service config');
+    this.actions_ = new Actions(
+        this.ampdoc_, this.urlBuilder_,
+        this.subscriptionAnalytics_,
+        this.validateActionMap(this.serviceConfig_['actions'])
+    );
+  }
+
+  /**
+   * Validates the action map
+   * @param {!JsonObject<string, string>} actionMap
+   * @returns {!JsonObject<string, string>}
+   */
+  validateActionMap(actionMap) {
+    user().assert(actionMap['login'],
+        'Login action is not present in action map');
+    user().assert(actionMap['subscribe'],
+        'Subscribe action is not present in action map');
+    return actionMap;
+  }
+
+  /**
+   * @return {!Promise<string>}
+   * @private
+   */
+  getReaderId_() {
+    if (!this.readerIdPromise_) {
+      const consent = Promise.resolve();
+      this.readerIdPromise_ = this.cid_.then(cid => {
+        return cid.get(
+            {scope: 'amp-subscriptions', createCookieIfNotPresent: true},
```

Comment:
For transition purposes, let's do `amp-access` here instead.

Reply:
Do we change this later?

Reply:
No. Probably will always keep it as this. It's just a string - as long as it "sounds like" for a related reason, we are all good.